dashboard/app/views/home/_homepage.html.haml

```diff
@@ -29,6 +29,7 @@
   - else
     - homepage_data[:sections] = @student_sections
     - homepage_data[:isTeacher] = false
+    - homepage_data[:studentId] = current_user.id
```

Comment:
My instinct here is: If we're passing an additional value down in the view anyway (in this case the user id), perhaps we should simply pass the count of unseen teacher feedbacks in the view instead, and avoid adding another asynchronous request to the page load (which could also cause this notification to "pop in" later than the rest of the page).

I'm not :100: confident in this instinct. Passing that value in the view tightly couples the view to this feature.

We've never established a consistent approach. I don't want more detailed knowledge of our client code's needs in this view, nor do I want more small API calls on page load. It's possible a more robust API like [GraphQL](https://graphql.org) is the path forward, where we've got a thin, cacheable view making one API call on load that asks for all of the data it needs to hydrate.  But do we build that now?

Interested in others' thoughts on this.

Reply:
i'm in the same boat as brad. on the client, we're only currently using the feedback count, so my instinct is to either 1) pass the feedback count through this view to the client, or 2) simplify the endpoint to return the feedback count (most likely the former since it removes an extra request to the server).

Reply:
I went with simplifying the endpoint to only return the count.  Since we need to use this notification in 3 places (the homepage, the unit overview and the course overview) it seemed handy if the notification could deal with retrieving the information it needs rather than threading the count to 3 different places. Seem reasonable?

Reply:
yes, i like that idea! thanks, erin 😄